azurelinuxagent/ga/update.py

```diff
@@ -562,6 +580,8 @@ def _process_goal_state(self, exthandlers_handler, remote_access_handler, agent_
         # check for agent updates
         agent_update_handler.run(self._goal_state, self._processing_new_extensions_goal_state())
 
+        self._wait_for_cloud_init()
```

Comment:
The wait I am adding is against the Final stage. PA does not wait for Final because there could be lengthy operations there that could affect TDPR (that is why the option I added needs to be explicitly set on a custom image).

```
https://cloudinit.readthedocs.io/en/latest/explanation/boot.html

There are five stages to boot:

1 Detect
2 Local
3 Network
4 Config
5 Final
```